packages/react-reconciler/src/ReactFiberCommitWork.new.js

```diff
@@ -3589,7 +3600,8 @@ function commitHookPassiveUnmountEffects(
   if (
     enableProfilerTimer &&
     enableProfilerCommitHooks &&
-    finishedWork.mode & ProfileMode
+    finishedWork.mode & ProfileMode &&
+    enableProfilingDEV
```

Comment:
I think what this `enableProfilingDEV` check is really about is whether we're executing these effects during the actual commit phase, because when the profiler logic was written, it (understandably) did not account for the possibility that effects could be invoked imperatively.

So now that I see this, I think what we could do instead of adding another variable is check whether we're in the commit phase.

There's a `executionContext` variable in the work loop that tells us this information:

https://github.com/facebook/react/blob/7bcc687720bbb31873768399dd87c27d8c231011/packages/react-reconciler/src/ReactFiberWorkLoop.new.js#L274-L275

So you should be able to replace this new variable with `executionContext & CommitContext`